RegistryManager/src/main/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryController.java

```diff
@@ -54,45 +52,94 @@ public void init() {
+				});
+
 	}
 
 	@Path("/{registrationId}")
 	@GET
 	public Uni<RestResponse<Object>> getCSourceById(HttpServerRequest request,
 			@PathParam("registrationId") String registrationId) {
 		logger.debug("get CSource() ::" + registrationId);
-		ArrayListMultimap<String, String> headers = HttpUtils.getHeaders(request);
-		return HttpUtils.validateUri(registrationId).onItem().transformToUni(t1 -> {
-			String tenant = HttpUtils.getTenantFromHeaders(headers);
-			return csourceService.getRegistrationById(registrationId, tenant).onItem().transformToUni(t -> {
-				return HttpUtils.generateReply(request, t, AppConstants.CSOURCE_URL_ID);
-			});
-		}).onFailure().recoverWithItem(HttpUtils::handleControllerExceptions);
+		int acceptHeader = HttpUtils.parseAcceptHeader(request.headers().getAll("Accept"));
+		if (acceptHeader == -1) {
+			return HttpUtils.INVALID_HEADER;
+		}
+
+		Context context;
+		List<Object> headerContext;
+		try {
+			HttpUtils.validateUri(registrationId);
+			headerContext = HttpUtils.getAtContext(request);
+			context = JsonLdProcessor.getCoreContextClone().parse(headerContext, false);
+		} catch (Exception e) {
+			return Uni.createFrom().item(HttpUtils.handleControllerExceptions(e));
+		}
+		return csourceService.retrieveRegistration(HttpUtils.getTenant(request), registrationId).onItem()
+				.transform(entity -> {
+					return HttpUtils.generateEntityResult(headerContext, context, acceptHeader, entity, null, null,null);
+				}).onFailure().recoverWithItem(HttpUtils::handleControllerExceptions);
+
 	}
 
 	@Path("/{registrationId}")
 	@PATCH
 	public Uni<RestResponse<Object>> updateCSource(HttpServerRequest request,
 			@PathParam("registrationId") String registrationId, String payload) {
-		return EntryControllerFunctions.appendToEntry(csourceService, request, registrationId, payload, "",
-				AppConstants.CSOURCE_REG_UPDATE_PAYLOAD, logger);
+		Tuple2<Context, Map<String, Object>> tuple;
+		try {
+			tuple = HttpUtils.expandBody(request, payload, AppConstants.CSOURCE_REG_UPDATE_PAYLOAD);
+		} catch (Exception e) {
+			return Uni.createFrom().item(HttpUtils.handleControllerExceptions(e));
+		}
+		return csourceService.updateRegistration(HttpUtils.getTenant(request), registrationId, tuple.getItem2())
+				.onItem().transform(opResult -> {
+					return HttpUtils.generateUpdateResultResponse(opResult);
+				}).onFailure().recoverWithItem(error -> {
+					return HttpUtils.handleControllerExceptions(error);
+				});
 	}
 
 	@Path("/{registrationId}")
 	@DELETE
 	public Uni<RestResponse<Object>> deleteCSource(HttpServerRequest request,
 			@PathParam("registrationId") String registrationId) {
-		return EntryControllerFunctions.deleteEntry(csourceService, request, registrationId, logger);
+		int acceptHeader = HttpUtils.parseAcceptHeader(request.headers().getAll("Accept"));
+		if (acceptHeader == -1) {
+			return HttpUtils.INVALID_HEADER;
+		}
+		try {
+			HttpUtils.validateUri(registrationId);
```

Comment:
Please verify validateUri method in case of entered wrong format registrationId.

Reply:
when registration Id format is wrong, then throw an exception link "id is not a URI".